### ISSUE TITLE
debug not working in 3.0.0-rc1

### ISSUE THREAD

**Issue by murali418** [phpredis/phpredis#943]
hi,
i recently updated php redis version to 3.0.0-rc1after that $redis->debug('key'); not working,
no errors

**Comment by yatsukhnenko** [comment]
No errors so everything is OK :smile:

**Comment by murali418** [comment]
hi @yatsukhnenko  thanks for the reply 
but still no results showing
$data = $redis->debug('Key');
var_dump($data); gives output  **bool(false)**
is debug method disabled or need to do anything to make that method work

**Comment by axel-rutz** [comment]
I can confirm this:
```
$c=\Drupal\redis\ClientFactory::getCLient();
$c->set('foo', 17);
$x=$c->debug('foo');
var_dump($x);
```
Result:
`bool(false)`

Expected: (something like)
`Value at:0x7f6d7b94af20 refcount:1 encoding:ziplist serializedlength:176 lru:13576901 lru_seconds_idle:1041`

**Comment by axel-rutz** [comment]
With the above key set, this is remarkable:
```
$x=$c->rawCommand('DEBUG', 'OBJECT', 'foo');
var_dump($x);
```
Result:
`bool(true)`

To the contrary:
```
$x=$c->rawCommand('DEBUG', 'OBJECT', 'non-existing-key');
var_dump($x);
```
Result:
`bool(false)`

**Comment by hhxsv5** [comment]
I got the same problem, changing the version to 2.x cannot work.

**Comment by hosef** [comment]
The test for the debug function was removed in 79cb6a8905ad8e2d2900e1b51db5fdaf37129632 during a refactoring of the unit tests.
The debug function itself was then later removed in 1f5aa6e8ed3c9852c451080e9017d1b8fb43e665.

**Comment by michael-grunder** [comment]
Hey everyone,

The reason phpredis returns true/false for debug is that the way we process simple string replies is to turn them into a boolean.  This is actually what most people want for virtually every other command (e.g. we don't return `"+OK"` but rather `true` for a set command).

There is a currently open pull request related to this problem that lets phpredis interact with Disque [here](https://github.com/phpredis/phpredis/pull/1243).

I think the way it works now is pretty clearly a bug but the question is how to properly handle it.  Only a tiny minority of people are going to want to see the _literal_ string response rather than the simpler boolean value for success and or failure so it certainly needs to be a non default option at the very least.

Looks like the code to process the debug response is still in place, so it shouldn't be too difficult to bring it back.  It does make me slightly nervous because of something like `DEBUG SEGFAULT` but I suppose one could already do that via `rawCommand` anyway so it wouldn't really add any additional danger.

**Comment by ilufei** [comment]
sorry，I want to know the bug fix or not ？

**Comment by michael-grunder** [comment]
This has been fixed via 5cb30fb2a65abc9d31ff58af6da0fbb6be4ec153.

You can now do something like this:

```php
<?php
$obj_r = new Redis();
$obj_r->connect('127.0.0.1', 6379);
$obj_r->set('foo', 'bar');
$obj_r->setOption(Redis::OPT_REPLY_LITERAL, true);
var_dump($obj_r->rawCommand('DEBUG', 'object', 'foo'));
```
